DM-41241: Create a new dev butler central repo in s3://rubin-pp-dev-users and move there (#120)
lsst-dm/prompt_processing: doc/playbook.rst

* ``rubin-pp-dev-users/unobserved/`` contains raw files that the upload script(s) can draw from to create incoming raws.

``rubin-pp-dev`` has had notifications configured for it; these publish to the Kafka topic ``prompt-processing-dev``.
The notifications can be viewed at `Kafdrop <https://k8s.slac.stanford.edu/usdf-prompt-processing-dev/kafdrop>`_.

> .

Reviewer: Extra period. Should the Kafdrop link have been included in the previous commit?

Author: IIRC the "previous commit" was the commit changing all old bucket names to the new ones. As the Kafdrop setup is independent of the bucket change, to me it looks clearer to keep one commit purely about the bucket change, and another commit about updating the playbook.

Reviewer: Ok, I thought it was the result of setting up the bucket.

> The buckets ``rubin:rubin-pp`` and ``rubin:rubin-pp-users`` are also for Prompt Processing development and previously used by the testers.

Reviewer: I think the "Buckets" section has grown long enough that it needs to be broken up into sections. Maybe "Current Buckets", "Legacy Buckets", and "AWS Credentials"?

Author: Great idea! The section is reorganized and rewritten in https://github.com/lsst-dm/prompt_processing/pull/120/commits/3d07119c17885cb8213e5bd7e998a99f23df9edf

``rubin:rubin-pp-users`` contains an older version of the development central repository.
``rubin:rubin-pp`` has notifications configured to publish to the Kafka topic ``rubin-prompt-processing``.

The Ceph user ``rubin-prompt-processing`` owns the ``rubin:rubin-pp`` and ``rubin:rubin-pp-users`` buckets, and the newer Ceph user ``prompt-processing-dev`` owns the ``rubin-pp-dev`` and ``rubin-pp-dev-users`` buckets.

> We are in the process to deprecate the ``rubin-prompt-processing`` user as it has a more restrictive permission than ``prompt-processing-dev``.

Reviewer: ```suggestion
We are in the process of deprecating the ``rubin-prompt-processing`` user as it has more restrictive permissions than ``prompt-processing-dev``.
```

The default Rubin users' setup on ``rubin-devl`` includes an AWS credential file at the environment variable ``AWS_SHARED_CREDENTIALS_FILE`` and a default profile without read permission to the prompt processing buckets.

> A separate credential for prompt processing developers as the Ceph user ``prompt-processing-dev`` (new) or ``rubin-prompt-processing`` (old) is at  `vault <https://vault.slac.stanford.edu/ui/vault/secrets/secret/show/rubin/usdf-prompt-processing-dev/s3-buckets>`_.

Reviewer: Mention that to get `rubin-prompt-processing` you need to look at an older version of the secret? (Or is there another way?)

   You must pass the ``--endpoint-url`` argument even if you have ``S3_ENDPOINT_URL`` defined.

> If a bucket is a Ceph tenant bucket and requires a tenant prefix, the bucket name violates the standard and is not supported by AWS CLI.

Reviewer: Say explicitly that these are the buckets that start with `rubin:`?

>     mc alias set usdf-pp https://s3dfrgw.slac.stanford.edu ACCESS_KEY SECRET_KEY

Reviewer: Should the instructions for AWS credentials also be updated to use a `prompt-processing-dev` profile? I find it less confusing if the profile name matches the user name.

    setup -r prompt_processing

The tester scripts send ``next_visit`` events for each detector via Kafka on the ``next-visit-topic`` topic.
They then upload a batch of files representing the snaps of the visit to the ``rubin-pp-dev`` S3 bucket, simulating incoming raw images.

> Eventually a set of parallel processes running on multiple nodes will be needed to upload the images sufficiently rapidly.

Reviewer: We already have this! 😁 (Well, we didn't need the "multiple nodes" part.)

Author: 👍  . Update this in a [separate commit ](https://github.com/lsst-dm/prompt_processing/pull/120/commits/a50fe44385d7e0003b506af0db16d0c891c71d03)
